pkg/detector/preemption.go

```diff
@@ -113,6 +120,28 @@ func (d *ResourceDetector) preemptPropagationPolicy(resourceTemplate *unstructur
 	return nil
 }
 
+// preemptClusterPropagationPolicyDirectly directly preempts resource template claimed by ClusterPropagationPolicy regardless of priority.
+func (d *ResourceDetector) preemptClusterPropagationPolicyDirectly(resourceTemplate *unstructured.Unstructured, policy *policyv1alpha1.PropagationPolicy) error {
+	claimedPolicyName := util.GetLabelValue(resourceTemplate.GetLabels(), policyv1alpha1.ClusterPropagationPolicyLabel)
+	if claimedPolicyName == "" {
+		return nil
+	}
+
+	clusterWideKey, err := keys.ClusterWideKeyFunc(resourceTemplate)
+	if err != nil {
+		// should not happen.
+		return err
+	}
+	if err := d.ApplyPolicy(resourceTemplate, clusterWideKey, policy); err != nil {
+		klog.Errorf("Failed to apply new propagation policy(%s/%s) on resource template(%s, kind=%s, %s) directly: %v.", policy.Namespace, policy.Name,
+			resourceTemplate.GetAPIVersion(), resourceTemplate.GetKind(), names.NamespacedKey(resourceTemplate.GetNamespace(), resourceTemplate.GetName()), err)
+		return err
+	}
```

Comment:
Can we just claim the resource template here? like:
```golang
	err = d.ClaimPolicyForObject(resourceTemplate, policy.Namespace, policy.Name)
	if err != nil {
		klog.Errorf("xxxx")
		return err
	}
```
This might be cheaper than `ApplyPolicy`, and just need to sync/update RB once. What do you think?